logging/access.go

```diff
@@ -109,6 +115,29 @@ func stripQueryString(u string) string {
 	}
 }
 
+// maskQueryParams masks (i.e., hashing) specific query parameters in the provided request's URI.
+// Returns the obfuscated URI.
+func maskQueryParams(req *http.Request, maskedQueryParams map[string]bool) string {
```

Comment:
I think we can also inline map len check here and return early

Reply:
But this function returns string, means we need to allocate new one even for early return.